src/Mapper/BaseGroupedMapper.php

```diff
@@ -187,10 +187,16 @@ public function ifFalse($bool)
     }
 
     /**
+     * @throws \LogicException
+     *
      * @return $this
      */
     public function ifEnd()
     {
+        if (empty($this->apply)) {
+            throw new \LogicException('No open ifTrue or ifFalse, you cannot use ifEnd()');
```

Comment:
Either add `()` to all methods or remove the last one

Reply:
Also please quote the methods

Reply:
I added `()`.

I didn't use quote because in the message "No open tabs or groups, you cannot use end()", the method is not quoted.